pulumi/infra/service_queue.py

```diff
@@ -19,45 +21,72 @@ def __init__(
+        message_retention_seconds = 60 * 60 * 24 * 4  # 4 days
 
         # `arn` is the ARN of a queue. This is a function because of
         # the need to use Output.apply on the ARN.
         def redrive_policy(arn: pulumi.Output) -> str:
             return json.dumps(
                 {
                     "deadLetterTargetArn": arn,
-                    "maxReceiveCount": 10,
+                    "maxReceiveCount": 3,
                 }
             )
 
         # TODO: delete_before_replace is only needed if we're
         # overriding the name of the queues
 
-        dead_letter_name = f"grapl-{name}-dead_letter-queue"
+        prefix = pulumi.get_stack()
+
+        # Queues have to be imported by URL, which includes the
+        # account ID
+        account_id = (
+            "000000000000" if IS_LOCAL else aws.get_caller_identity().account_id
+        )
+        queue_import_prefix = f"https://queue.amazonaws.com/{account_id}"
+
+        logical_dead_letter_name = f"{name}-dead-letter-queue"
+        physical_dead_letter_name = f"{prefix}-{logical_dead_letter_name}"
         self.dead_letter_queue = aws.sqs.Queue(
-            dead_letter_name,
-            name=dead_letter_name,
+            logical_dead_letter_name,
+            name=physical_dead_letter_name,
             message_retention_seconds=message_retention_seconds,
-            opts=pulumi.ResourceOptions(parent=self, delete_before_replace=True),
+            visibility_timeout_seconds=30,
+            opts=util.import_aware_opts(
+                f"{queue_import_prefix}/{physical_dead_letter_name}",
+                parent=self,
+                delete_before_replace=True,
+            ),
         )
 
-        retry_name = f"grapl-{name}-retry-queue"
+        logical_retry_name = f"{name}-retry-queue"
+        physical_retry_name = f"{prefix}-{logical_retry_name}"
         self.retry_queue = aws.sqs.Queue(
-            retry_name,
-            name=retry_name,
+            logical_retry_name,
+            name=physical_retry_name,
             message_retention_seconds=message_retention_seconds,
+            visibility_timeout_seconds=360,
             redrive_policy=self.dead_letter_queue.arn.apply(redrive_policy),
-            opts=pulumi.ResourceOptions(parent=self, delete_before_replace=True),
+            opts=util.import_aware_opts(
+                f"{queue_import_prefix}/{physical_retry_name}",
+                parent=self,
+                delete_before_replace=True,
+            ),
         )
 
-        queue_name = f"grapl-{name}-queue"
+        logical_queue_name = f"{name}-queue"
+        physical_queue_name = f"{prefix}-{logical_queue_name}"
         self.queue = aws.sqs.Queue(
-            queue_name,
-            name=queue_name,
+            logical_queue_name,
+            name=physical_queue_name,
             message_retention_seconds=message_retention_seconds,
+            visibility_timeout_seconds=180,
```

Comment:
I'll file an issue to follow up on this, since it doesn't seem like there's a straightforward place in that code where that assumption is encoded. These numbers are basically coming straight from the CDK code as it exists now.

Reply:
https://github.com/grapl-security/issue-tracker/issues/333